src/Analyser/NodeScopeResolver.php

```diff
@@ -1384,7 +1384,10 @@ private function processStmtNode(
 				$loopScope = $this->processExprNode($stmt, $loopExpr, $loopScope, $nodeCallback, ExpressionContext::createTopLevel())->getScope();
 			}
 			$finalScope = $finalScope->generalizeWith($loopScope);
+
+			$alwaysIterates = TrinaryLogic::createFromBoolean($context->isTopLevel());
```

Comment:
Would something fail if you said `$alwaysIterates = TrinaryLogic::createYes()` here?

Reply:
no, apparently not. this comes basically from the other loops which have code like the following
```php
$condBooleanType->isTrue()->yes() && $context->isTopLevel()
```

but tbh, I didn't fully understand what the level here changes or does

Reply:
OK, makes sense to be consistent.

This condition asks about whether it's the "analysis iteration", or one of the "let's find out if types stabilize in the loop before we generalize them" that's done for all loops in a do-while like this: https://github.com/phpstan/phpstan-src/blob/c3bb0557b3b34770e9a68cf7f5f5cc3c1f6291fd/src/Analyser/NodeScopeResolver.php#L1330-L1360